source/expectations/data_values/ut_compound_data_helper.pkb

```diff
@@ -56,236 +43,473 @@ create or replace package body ut_compound_data_helper is
+      end if;
+    exit when (a_data_info.access_path = l_pk_tab(l_index)) or l_index = l_pk_tab.count;
+    l_index := l_pk_tab.next(l_index);
+    end loop;
+    ut_utils.append_to_clob(a_join_by_stmt,l_sql_stmt);
+    end if;
+  end;
+  
+  procedure generate_equal_sql(a_equal_stmt in out nocopy clob,a_col_name in varchar2) is  
+    l_sql_stmt varchar2(32767);
+  begin
+    l_sql_stmt := case when a_equal_stmt is null then null else ' and ' end ||' a.'||a_col_name||q'[ = ]'||' e.'||a_col_name;
+    ut_utils.append_to_clob(a_equal_stmt,l_sql_stmt);
+  end;
+
+  procedure generate_partition_stmt(
+    a_data_info ut_cursor_column, a_partition_stmt in out nocopy clob,
+    a_pk_table in ut_varchar2_list,a_col_name in varchar2,a_alias varchar2 := 'ucd.'
+  ) is
+    l_alias varchar2(10) := a_alias;
+    l_pk_tab ut_varchar2_list := coalesce(a_pk_table,ut_varchar2_list());
+    l_index integer;
+    l_sql_stmt varchar2(32767);
+  begin    
+    if l_pk_tab.count <> 0 then
+      l_index:= l_pk_tab.first;
+      loop
+        if l_pk_tab(l_index) in (a_data_info.access_path, a_data_info.parent_name) then
+          --When then table is nested and join is on whole table
+          l_sql_stmt := case when a_partition_stmt is null then null else ',' end;
+          l_sql_stmt := l_sql_stmt ||l_alias||a_col_name;
+        end if;
+        
+        exit when (a_data_info.access_path = l_pk_tab(l_index)) or l_index = l_pk_tab.count;
+        l_index := l_pk_tab.next(l_index);
+      end loop;
+    else
+      l_sql_stmt :=  case when a_partition_stmt is null then null else ',' end ||l_alias||a_col_name;
+    end if;
+    ut_utils.append_to_clob(a_partition_stmt,l_sql_stmt); 
+  end;   
+  
+  procedure generate_select_stmt(a_data_info ut_cursor_column,a_sql_stmt in out nocopy clob, a_col_name varchar2,a_alias varchar2 := 'ucd.') is
+    l_alias varchar2(10) := a_alias;
+    l_col_syntax varchar2(4000);
+    l_ut_owner varchar2(250) := ut_utils.ut_owner;
+  begin    
+    if a_data_info.is_sql_diffable = 0 then 
+      l_col_syntax :=  l_ut_owner ||'.ut_compound_data_helper.get_hash('||l_alias||a_col_name||'.getClobVal()) as '||a_col_name ;
+    else 
+      l_col_syntax :=  l_alias||a_col_name||' as '|| a_col_name;
+    end if;   
+    ut_utils.append_to_clob(a_sql_stmt,','||l_col_syntax);
+  end;
+      
+  procedure generate_xmltab_stmt(a_data_info ut_cursor_column,a_sql_stmt in out nocopy clob, a_col_name varchar2) is
+    l_sql_stmt varchar2(32767);
+    l_col_type varchar2(4000);
+  begin    
+    if a_data_info.is_sql_diffable = 0 then 
+      l_col_type := 'XMLTYPE';
+    elsif a_data_info.is_sql_diffable = 1  and a_data_info.column_type = 'DATE' then 
+      l_col_type := 'TIMESTAMP';
+    elsif  a_data_info.is_sql_diffable = 1  and a_data_info.column_type in ('TIMESTAMP','TIMESTAMP WITH TIME ZONE') then
+      l_col_type := a_data_info.column_type;
+    --TODO : Oracle bug : https://community.oracle.com/thread/1957521
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this "TODO" comment. ([plsql:S1135](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=801&issues=AWd_NeYMCh2zBQVNm3C5&open=AWd_NeYMCh2zBQVNm3C5)

[ISSUE_KEY]: AWd_NeYMCh2zBQVNm3C5